### PR TITLE
[EXOD-020] Improve general lagginess in app

### PR DESCRIPTION
**Pull request by melondrone**
## Changes
- Fix margin inconsistencies on bond page (screeenshot below)
- Modify App/Bond/Account slices to use `Promise.all` to asynchronously make requests which aren't co-dependant. Should result in a quicker load time for the app.
  - This was tested locally using the docker build as an A/B test before and after. The changes seem to have resulted in a 1 second load time improvement.
- Load the Graph Data separately from the App data (no need to load them at the same time and graph data can be loaded earlier as it isn't dependant on a web3 connection)
- Improve the "Pending" states of several claim/bond button functions
- Reduce the debounce of the purchase bond page from 1 second to 0,5 seconds. Resulting in a more smooth user experience.
- Add dynamic block rate seconds calculation and use this in in all areas which previously used the constant value (currently it's still 0.9 seconds)
- Remove unused file (`CalcChart.tsx`)

Screenshot from 2021-12-09 22-35-47
Screenshot from 2021-12-09 22-36-10

### CODE REVIEW

File: src/slices/AccountSlice.ts [hunk 1]
```diff
@@ -42,11 +40,12 @@ export const getBalances = createAsyncThunk(
       provider,
     ) as IERC20;
     const poolBalance = await poolTokenContract.balanceOf(address);
-*/
+    */
+    const results = await Promise.all([ohmContract.balanceOf(address), sohmContract.balanceOf(address)]);
```
Comment (HOVOH) [inline src/slices/AccountSlice.ts:44]: When calling a function which return an arrays such as the Promise.all function you can use the following syntax to get named variables, which will help code readability:
`const [ohm, sohm] = await Promise.all(...)`

File: src/slices/AccountSlice.ts [hunk 2]
```diff
@@ -70,21 +69,23 @@ export const loadAccountDetails = createAsyncThunk(
...
+      ohmContract.allowance(address, addresses[networkID].STAKING_HELPER_ADDRESS),
+      sohmContract.allowance(address, addresses[networkID].STAKING_ADDRESS),
+    ]);
+    dispatch(getBalances({ address, networkID, provider }));
 
     return {
       staking: {
-        ohmStake: +stakeAllowance,
-        ohmUnstake: +unstakeAllowance,
+        ohmStake: +results[0],
```
Comment (HOVOH) [inline src/slices/AccountSlice.ts:87]: Same thing here

File: src/slices/AccountSlice.ts [hunk 3]
```diff
@@ -126,18 +127,27 @@ export const calculateUserBondDetails = createAsyncThunk(
...
+      balance = BigNumber.from(0);
+
+    const results = await Promise.all([
+      bondContract.bondInfo(address),
+      bondContract.pendingPayoutFor(address),
+      reserveContract.allowance(address, bond.getAddressForBond(networkID)),
+      reserveContract.balanceOf(address),
+    ]);
+
+    const bondDetails = results[0];
```
Comment (HOVOH) [inline src/slices/AccountSlice.ts:140]: and here

File: src/slices/AppSlice.ts [hunk 4]
```diff
@@ -100,37 +42,95 @@ export const loadAppDetails = createAsyncThunk(
...
     ) as SOhmv2;
 
+    const results = await Promise.all([
+      provider.getBlockNumber(),
+      stakingContract.epoch(),
+      sohmMainContract.circulatingSupply(),
+      stakingContract.index(),
+    ]);
+
+    const currentBlock = results[0];
```
Comment (HOVOH) [inline src/slices/AppSlice.ts:52]: here

**Review by HOVOH — COMMENTED** [review]
Amazing job! Just start using the const [var1, var2] syntax and I'm merging

**Comment by melondrone** [comment]
> Just start using the const [var1, var2] syntax and I'm merging

Good shout! I've changed it where i can :)

**Pull request merged**